lib/stringify.js

```diff
@@ -816,27 +881,19 @@ compilerPrototype.heading = function (node) {
  * @param {Object} node - `text` node.
  * @return {string} - Raw markdown text.
  */
-compilerPrototype.text = function (node) {
-    return this.encode(node.value, node);
-};
+compilerPrototype.text = function (node, parent) {
+    var value = this.encode(node.value, node);
 
-/**
- * Stringify escaped text.
- *
- * @example
- *   var compiler = new Compiler();
- *
- *   compiler.escape({
- *     type: 'escape',
- *     value: '\n'
- *   });
- *   // '\\\n'
- *
- * @param {Object} node - `escape` node.
- * @return {string} - Markdown escape.
- */
-compilerPrototype.escape = function (node) {
-    return '\\' + node.value;
+    if (
+        value[value.length - 1] == '\\' && parent &&
+        parent.type == 'linkReference' &&
+        parent.referenceType == 'shortcut'
+    ) {
+        // Don't escape trailing backslash in shortcut links.
```

Comment:
I think you’re right, commonmark [disagrees](http://spec.commonmark.org/dingus/?text=%5C%5Ba%5D%20and%20%5Bb%5C%5D%0A%0A%5Ba%5D%3A%20http%3A%2F%2Fgoogle.com%0A%5Bb%5D%3A%20http%3A%2F%2Fgoogle.com) with mdast, and so does GH!